wandb/sdk/wandb_init.py

```diff
@@ -335,148 +247,64 @@ def _log_setup(self, settings):
-        self.backend.interface.send_output(name, data)
-
-    def _redirect(self, stdout_slave_fd, stderr_slave_fd):
-        console = self.settings.console
-        logger.info("redirect: %s", console)
-
-        if console == "redirect":
-            logger.info("redirect1")
-            out_cap = redirect.Capture(name="stdout", cb=self._redirect_cb)
-            out_redir = redirect.Redirect(
-                src="stdout", dest=out_cap, unbuffered=True, tee=True
-            )
-            err_cap = redirect.Capture(name="stderr", cb=self._redirect_cb)
-            err_redir = redirect.Redirect(
-                src="stderr", dest=err_cap, unbuffered=True, tee=True
-            )
-            try:
-                out_redir.install()
-                err_redir.install()
-                self._out_redir = out_redir
-                self._err_redir = err_redir
-                logger.info("redirect2")
-            except (OSError, AttributeError) as e:
-                logger.error("failed to redirect", exc_info=e)
-            return
-
-        return
-
-        # redirect stdout
-        if platform.system() == "Windows":
-            win32_redirect(stdout_slave_fd, stderr_slave_fd)
-        else:
-            self._save_stdout = sys.stdout
-            self._save_stderr = sys.stderr
-            stdout_slave = os.fdopen(stdout_slave_fd, "wb")
-            stderr_slave = os.fdopen(stderr_slave_fd, "wb")
-            stdout_redirector = io_wrap.FileRedirector(sys.stdout, stdout_slave)
-            stderr_redirector = io_wrap.FileRedirector(sys.stderr, stderr_slave)
-            stdout_redirector.redirect()
-            stderr_redirector.redirect()
-            self.stdout_redirector = stdout_redirector
-            self.stderr_redirector = stderr_redirector
-        logger.info("redirect done")
-
-    def _restore(self):
-        logger.info("restore")
-        # TODO(jhr): drain and shutdown all threads
-        if self._use_redirect:
-            if self._out_redir:
-                self._out_redir.uninstall()
-            if self._err_redir:
-                self._err_redir.uninstall()
-            return
-
-        if self.stdout_redirector:
-            self.stdout_redirector.restore()
-        if self.stderr_redirector:
-            self.stderr_redirector.restore()
-        if self._save_stdout:
-            sys.stdout = self._save_stdout
-        if self._save_stderr:
-            sys.stderr = self._save_stderr
-        logger.info("restore done")
+        self._wl._early_logger_flush(logger)
 
     def init(self):
         s = self.settings
         config = self.config
 
+        if s.reinit:
+            if len(self._wl._global_run_stack) > 0:
+                if len(self._wl._global_run_stack) > 1:
+                    print(
+                        "If you want to track multiple runs concurrently in wandb you should use multi-processing not threads"  # noqa: E501
+                    )
```

Comment:
Do we want a `termwarn` here?  In regular usage of reinit, this case shouldn't be hit right?

Reply:
fixed